packages/user-interface/src/components/Task.tsx

```diff
@@ -5,41 +5,94 @@ import useTaskUrl from "../hooks/useTaskUrl";
 import useActionLabels from "../hooks/useActionLabels";
 import { ButtonLink } from "@gemeente-denhaag/button-link";
 import { ChevronRightIcon } from "@gemeente-denhaag/icons";
+import useOgonePayment from "../hooks/useOgonePayment";
+import { Button } from "@gemeente-denhaag/components-react";
+import { useContext } from "react";
+import { LocaleContext } from "@nl-portal/nl-portal-localization";
+import { useHref, useLinkClickHandler } from "react-router-dom";
 
 interface Props {
   task: TaakV2;
+  openInContext?: boolean;
 }
 
-const Task = ({ task }: Props) => {
+const Task = ({ task, openInContext }: Props) => {
   const labels = useActionLabels();
-  const taskUrl = useTaskUrl(task);
+  const { currentLocale } = useContext(LocaleContext);
+  const { startPayment, renderPaymentRedirectForm } = useOgonePayment();
+  const taskUrl = useTaskUrl(task, openInContext);
+  const taskHref = useHref(taskUrl ?? "");
+  const handleClick = useLinkClickHandler(taskHref);
```

Comment:
Wat voeg dit toe aan de click? of is het gewoon een push history?

Reply:
Nu ik het zo zie is die useHref misschien niet nodig. Die useLinkClickHandler zorgt er voor dat de Utrecht ButtonLink geen page refresh doet, maar zich net zo gedraagd als de PortalLink. De PortalLink kunnen we hier echter niet gebruiken, omdat de ButtonLink geen component prop accepteert.